include/internal/string.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef HEADER_INTERNAL_STRING_H
+# define HEADER_INTERNAL_STRING_H
+
+# include <string.h>
+# include <malloc.h>
+
+# ifdef OPENSSL_SYS_WINDOWS
+
+#  define strcasecmp _stricmp
+#  define strncasecmp _strnicmp
+
+#  ifdef _WIN64
+#   define strlen(s) OPENSSL_strlen31(s)
+unsigned int OPENSSL_strlen31(const char *str);
+#  endif
+#  if defined(_MSC_VER) && _MSC_VER<=1200 && defined(_MT) && defined(isspace)
+   /* compensate for bug in VC6 ctype.h */
+#   undef isspace
+#   undef isdigit
+#   undef isalnum
+#   undef isupper
+#   undef isxdigit
+#  endif
```

Comment:
Put the `is*` defines in `internal/ctype.h`

Reply:
Two things in the context. One can't compile OpenSSL with _MSC_VER<=1200 anymore, so that this #if section can be removed. Secondly, libcrypto relies on internal/ctype.h now, so it can be double-removed.

[Just in case for reference. What are these undefs about? They are about same thing as redefinitions of e.g. stat as _stat. It's about producing binary .obj files that can be linked into either .dll or .exe with either /Mx flag. They got it right for other iswhatever-s but not for these.]

Reply:
See #4263